src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductInput\BasicInformationInput;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductInput\ProductOptionsInput;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+//@todo: abit confusing when it should always build only one command but is called plural and returns array. smth needs adjustments
+class UpdateProductCommandsBuilder implements MultiShopProductCommandsBuilderInterface
+{
+    public function buildCommands(ProductId $productId, array $formData, ShopConstraint $shopConstraint): array
+    {
+        $updateProductCommand = new UpdateProductCommand($productId, $shopConstraint);
+
+        $updateProductCommand
+            ->setBasicInformation($this->buildBasicInfo($formData))
+            ->setOptions($this->buildOptions($formData))
+        ;
+
+        return [$updateProductCommand];
+    }
+
+    private function buildBasicInfo(array $formData): ?BasicInformationInput
```

Comment:
the class will grow into something having like 5-6 of similar methods. Idk if thats a problem? (it has no potential to grow alot more :thinking:)

Reply:
You're write it could be very verbose. In an other hand, I don't see another way to have right return Type. For a first implementation I would leave it like this.

Reply:
I changed my mind. See my review for more details.